File: SozlukJava/app/src/main/java/com/example/sozlukjava/viewmodel/EntryEkleViewModel.java
package com.example.sozlukjava.viewmodel;


import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.sozlukjava.dao.SozlukDao;
import com.example.sozlukjava.model.SimpleResponse;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class EntryEkleViewModel extends ViewModel {
    private final SozlukDao dao = new SozlukDao();
    private final MutableLiveData<SimpleResponse> addResult = new MutableLiveData<>();

    public MutableLiveData<SimpleResponse> getAddResult() {
        return addResult;
    }

    public void addEntry(int userId, String title, String content) {
        dao.addEntry(userId, title, content).enqueue(new Callback<SimpleResponse>() {
            @Override
            public void onResponse(Call<SimpleResponse> call, Response<SimpleResponse> response) {
                addResult.setValue(response.body());
            }

            @Override
            public void onFailure(Call<SimpleResponse> call, Throwable t) {
                SimpleResponse sr = new SimpleResponse();
                sr.setSuccess(false);
                sr.setMessage("Bağlantı hatası");
                addResult.setValue(sr);
            }
        });
    }
}

File: SozlukKotlin/app/src/main/java/com/example/sozlukkotlin/viewmodel/EntryEkleViewModel.kt
package com.example.sozlukkotlin.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.sozlukkotlin.dao.SozlukDao
import com.example.sozlukkotlin.model.SimpleResponse
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class EntryEkleViewModel : ViewModel() {
    private val dao = SozlukDao()
    val addResult = MutableLiveData<SimpleResponse>()

    fun addEntry(userId: Int, title: String, content: String) {
        dao.addEntry(userId, title, content).enqueue(object : Callback<SimpleResponse> {
            override fun onResponse(call: Call<SimpleResponse>, response: Response<SimpleResponse>) {
                addResult.value = response.body()
            }

            override fun onFailure(call: Call<SimpleResponse>, t: Throwable) {
                addResult.value = SimpleResponse(false, "Bağlantı hatası")
            }
        })
    }
}
